lib/cosmos/script/tools.rb

```diff
@@ -69,12 +69,22 @@ def set_stdout_max_lines(max_lines)
 
     def step_mode
       if defined? ScriptRunnerFrame
+        _ensure_script_runner_frame do
+          Qt.execute_in_main_thread do
+            ScriptRunnerFrame.instance.toggle_debug(true)
+          end
+        end
         ScriptRunnerFrame.step_mode = true
       end
     end
 
     def run_mode
       if defined? ScriptRunnerFrame
+        _ensure_script_runner_frame do
```

Comment:
I don't think run_mode() needs to remove the debug prompt.  It could be there for other reasons.